kebechet/services.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+# Kebechet
+# Copyright(C) 2018, 2019 Kevin Postlethwait
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Provides abstraction of service utility functions."""
+
+
+import requests
+import tempfile
+import logging
+
+from .enums import ServiceType
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _keep_slug(slug: str):
+    return slug
+
+
+def _encode_fslash(slug: str):
+    return slug.replace("/", "%2F")
+
+
+# NOTE: config_url could be generated using construct_raw_file_url but I feel this makes the code easier to follow
+services = {
```

Comment:
I would make `services` (private class var `_SERVICES`), `_encode_fslash` and `_keep_slug` (private methods) part of `Service`. They semantically corresponded.